### DISCUSSION TITLE
mcache

### DISCUSSION THREAD

**Discussion by alkkkj** [hse-project/hse#49] [Q&A]
I am looking use a user space file system as backend of HSE, but mcache logical cause things hardly.  For a user space file system, do not need use mmap function to access memory in kernel space. 

Whether can intergrating user space file system just using open/read/write/close interface?

**Comment by nabeelmmd** [comment]
What user space filesystem are you planning to use? We have not tested HSE against user space filesystems.

We currently do not have a way to disable mmap() usage in HSE. Our cursor implementation heavily relies on mcache.

**Comment by alkkkj** [comment]
Ok thanks,  I am interested in making things run on spdk/blobfs.
Maybe I can try to use userfaultfd to get mmap implemented in user space,
but not sure the impact on performance.

Nabeel ***@***.***> 于2021年10月6日周三 上午1:25写道：
